### DISCUSSION TITLE
Creating a new line after adding custom DecoratorBlockNode

### DISCUSSION THREAD

**Discussion by xenobytezero** [facebook/lexical#5773] [Q&A] [answered]
I have created a new plugin/node that subclasses DecoratorBlockNode, and noticed that after I add my new node, a new paragraph element after it is not created, when it is for a bunch of other similar nodes in the Playground (Figma, YouTube, etc). The node simply gets created and it's impossible to add any more content afterwards because there is no way to make a new paragraph.

My node doesn't seem that different in structure from the above nodes, so I assume I'm missing something.

Is there some specific configuration or structure I'm missing?

**Comment by ivailop7** [comment]
I think you a probably missing a property like 'isInline': false or one of those that triggers the behaviour

**Comment by xenobytezero** [comment]
That was my guess too, but it looks like `DecoratorBlockNode` already has that set to false.

https://github.com/facebook/lexical/blob/111117d0013917d0481cf418430db75a8afbb497/packages/lexical-react/src/LexicalDecoratorBlockNode.ts#L55-L57

**Comment by ivailop7** [comment]
Reread the question and realized, I misunderstood it originally. To add a paragraph on creation, you have to do this manually in the insert command for your node. You can create the node there and append a paragraph after it in there.

**Comment by xenobytezero** [comment, marked as answer]
That's what I had assumed, I just couldn't figure out how the Playground nodes were doing it, turns out the functionality is indeed in the commands rather than the node/component. Specifically, the use of `$insertNodeToNearestRoot`, which will do this for you.

Thanks for the pointers!
